### DISCUSSION TITLE
How can you add new nodes to a graph using a layout without resetting the positions of all other nodes?

### DISCUSSION THREAD

**Discussion by steveHornseyDeveloper** [antvis/G6#3403] [Q&A]
I'm working on a graph explorer where you start with a single node and add additional nodes to the original node. The data is directed with a maximum of two edges between each pair of nodes (one for each direction). I'd like to use a dagre layout for this graph, however, there is a requirement that every time a user moves a node, that position should be remembered.
I haven't found a way to add nodes using the dagre layout in a way which positions the new nodes according to the dagre layout without overriding the position of any node a user moved.

The two ways I have been trying to do this are
Using a .changeData() to add new nodes and passing the previous position of all nodes using the preset property.

https://g6.antv.vision/zh/docs/api/graphLayout/dagre#layoutcfgpreset
https://stackblitz.com/edit/react-hkcunr-szq8dg

The preset seems to be ignored and any node which was moved gets reverted to the algorithm's location. see #3392 

The other way I've tried is by using a sublayout of the existing node and the nodes to be added. However using a sublayout always positions the nodes in the top left of the graph, even when I specify a 'begin' property.

https://g6.antv.vision/zh/docs/api/graphLayout/dagre#layoutcfgbegin
https://stackblitz.com/edit/react-hkcunr-puekod

I can't seem to get begin to work at all even in a basic example #3402 

Does anyone know of a way to use a layout to add new nodes to a graph without resetting the position of any nodes a user has moved?

**Comment by steveHornseyDeveloper** [comment]
I've just noticed in my sandbox I was using g6 4.5.0 rather than 4.5.1, I've updated the dependency and https://stackblitz.com/edit/react-hkcunr-puekod now works 🙈 .

I'm still curious, does anyone else do something similar, and how?

**Reply by bsansone** [reply]
Hi @steveHornseyDeveloper did you ever find a solution to this?

**Comment by steveHornseyDeveloper** [comment]
Not really, the way we do this is that we calculate where we want the new node's position to be ourselves and then add it to the g6 graph.

I think we in a different usecase we use the g6 subgraph layouts method to calculate new positions for many nodes and then add the whole new sub graph to our canvas

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by bsansone** [reply]
@steveHornseyDeveloper How do you compute the new nodes position? I'm trying to achieve something similiar, but with a group of nodes. I was thinking of using the `antv/layout` library
